ts/Core/Axis/Axis.ts

```diff
@@ -3403,6 +3403,11 @@ class Axis {
                 Math.max((clipOffset as any)[invertedSide], clip);
         }
 
+        if (coll === 'colorAxis') {
+            this.chart.legend.render();
+            chart.getMargins(true);
+        }
+
```

Comment:
In order to avoid polluting the core with code dealing with ColorAxis, check if this code could be in an event handler for `afterGetOffset`. Then that event handler would be added from the ColorAxis module.